compile.sh

```diff
@@ -212,7 +212,7 @@ do
         echo "Invalid number of cores was specified: $2"
         exit 1
       fi
-      if [ "$MAXCORES" -ge "$2" -o "$MAXCORES" -eq 0 ]; then
+      if [ "$MAXCORES" -ge "$2" -o "$MAXCORES" -eq 0 -o -n "$DISTCC_POTENTIAL_HOSTS" -o -n "$DISTCC_HOSTS" ]; then
```

Comment:
Unfortunately I don't think this way won't be very robust. Neither are required to be set. For example, on my system, Manjaro, when distcc is installed, symlinks are setup in /usr/lib/distcc and I add that to my PATH, and then the hosts are read from ~/.distcc/hosts

Maybe someone from @distcc would have a suggestion about this @themightyoarfish @sourcefrog @shawnl @asheplyakov

Reply:
If it gets any more complicated then you should move distcc detection next to maxcores detection.

Reply:
👋🏻 I have never seen Widelands before (though it looks neat) and have never built it, so I'll just answer in general terms:

It looks like the point of this change is to block the user from scheduling more jobs than they have local cores. I'd just drop that! As @andy5995 said in the first comment. If the user wants 20 jobs on a 16 core machine, let them! I probably wouldn't add this complication into my own project. (In fact that's not hypothetical: I recently added a `-j` option to cargo-mutants and it doesn't check NCPUS: https://github.com/sourcefrog/cargo-mutants/pull/93) 

As @tothxa says, the number of cores is anyhow a poor proxy for "how many tasks can you usefully or safely run in parallel" because RAM, RAM bandwidth, and even IO bandwidth can be more important constraints. Also, at least some tools like Mold will spawn multiple threads internally.

Potentially naive users will overload their machine by running `-j 20000` but that's really on them; it's hard to reliably prevent it.

Also, distcc is now pretty old and there are newer and possibly better programs that carry on similar ideas. So I would avoid hardcoding anything specific to distcc.

Reply:
Hi @sourcefrog! Yes, this is exactly the info we needed. Thank you very much! 

So, this means the current approach to allow an user override for the jobs count check is OK.

The check should be kept though by default, because this script is mostly intended for users who'd like to help test pre-releases, and I think it's a reasonable rule of thumb to protect them.

Reply:
If you're saying you will use NCPUS-1 by default but let the user set whatever number they want then that sounds good to me: a good default but avoiding too much complication, and letting people set whatever number if they think they know best.

Reply:
Thanks for the feedback @sourcefrog . That code is outdated. I've removed the DISTCC-specific stuff, and the intent of this PR is to simply allow an override option.

@tothxa @Noordfrees I believe this is ready for review.